src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/config/web/CourtHearingEventFieldsFilter.kt

```diff
@@ -0,0 +1,97 @@
+package uk.gov.justice.digital.hmpps.courthearingeventreceiver.config.web
+
+import com.jayway.jsonpath.Configuration
+import com.jayway.jsonpath.JsonPath
+import com.jayway.jsonpath.PathNotFoundException
+import com.jayway.jsonpath.ReadContext
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.courthearingeventreceiver.service.TelemetryEventType
+import uk.gov.justice.digital.hmpps.courthearingeventreceiver.service.TelemetryService
+import java.util.stream.Collectors
+import javax.servlet.Filter
+import javax.servlet.FilterChain
+import javax.servlet.ServletRequest
+import javax.servlet.ServletResponse
+import javax.servlet.http.HttpServletRequest
+
+@Component
+class CourtHearingEventFieldsFilter(
+  @Autowired private val telemetryService: TelemetryService,
+  @Autowired private val observedFields: ObserveFields
+) : Filter {
+
+  override fun doFilter(request: ServletRequest?, response: ServletResponse?, filterChain: FilterChain?) {
+
+    val httpRequest = request as HttpServletRequest
+    if (observedFields.fields?.isNotEmpty()!!) {
+      val requestWrapper = CustomHttpRequestWrapper(httpRequest)
+      filterChain?.doFilter(requestWrapper, response)
+      val jsonContext: ReadContext = JsonPath.parse(requestWrapper.inputStream)
+      try {
+        trackEvent(jsonContext.jsonString(), observedFields)
+      } catch (exception: UnsupportedOperationException) {
+        return
+      }
+
+      return
+    }
+    filterChain?.doFilter(request, response)
+  }
+
+  private fun trackEvent(requestJson: String, observedFields: ObserveFields) {
+    val (defenceOrganisationAttributes, pncIdExist, croNumberExist) = buildEventDetails(requestJson, observedFields)
```

Comment:
This isn't quite what I'd envisaged, are you still working on this?  The intention was that we would just iterate over all of the fields in `observedFields.fields`, check what's at the PATH and then log that against the key provided in the config. This class shouldn't have any knowledge of any of the specific fields it's checking like defenceOrganisation or pnc. 

I'd expect this method to look more like:
```
var eventProperties = buildEventDetails(requestJson, observedFields)
    telemetryService.trackEvent(
      TelemetryEventType.COMMON_PLATFORM_EVENT_OBSERVED,
      eventProperties
    )
```

See later comment for `buildEventDetails` would work

Reply:
Got it. Will change the implementation.